fhir-server/src/main/java/com/ibm/fhir/server/resources/FHIRResource.java

```diff
@@ -419,10 +422,11 @@ protected String getRequestBaseUri(String type) throws Exception {
 
         // Strip off any path elements after the base
         if (type != null && !type.isEmpty()) {
-            int resourceNamePathLocation = baseUri.lastIndexOf("/" + type);
```

Comment:
this is the key change.  see the commit description for considerations

Reply:
what happens if this is hosted under `/Patient` or some other resource type. 
The intent of lastIndexOf was to be resilient to that approach. 

I think this is a trading off one approach for another which has similar issues. 

What happens if the the Logical ID is `Patient` in the code to the right? Wouldn't that exhibit a similar issue? 

Or if the server was hosted at https://server.me/PatientData/api/v4/Patient

The best way to harden this would be to pass in the JAXRS URL Path Segments and the JAXRS Query parameters and use these at this level to establish a resourceNamePathLocation as that will be clearly delineated among the  various options.   

This can be quickly packed into the FHIRRequestContext in the Jaxrs Filter

Reply:
> What happens if the the Logical ID is Patient in the code to the right? Wouldn't that exhibit a similar issue?

the proposed update here addresses exactly this issue (and even adds a test case for it)

> Or if the server was hosted at https://server.me/PatientData/api/v4/Patient

the propose update makes us flaky if someone changes the base path in a way such as this.  the difference is this base bath is controlled by the operator and not end users, and so i think its a much better option than what we have today


> I think this is a trading off one approach for another which has similar issues.

for sure.  i think its much better with the proposed change but its still just heuristic.  we can discuss if its worth doing something more robust tomorrow

Reply:
I'd ask that you please add documentation stating the limitation or add a test case showing that we don't support it.

Reply:
OK, I added a test and the following documentation to the sections on `X-FHIR-FORWARDED-URL` and `fhirServer/core/externalBaseUrl`:

> Note that the base URL must not include a path segment that matches any FHIR resource type name (case-sensitive). For example, "https://example.com" or "https://example.com/my/patient/api" are fine, but "https://example.com/my/Patient/api" is not.